### ISSUE TITLE
Mismatched type error in batch causes points to get registered but not written

### ISSUE THREAD

**Issue by sparrc** [influxdata/influxdb#7815]
### Bug report

When writing point batches that contain a mismatched type, some points get registered in the index but not written.

This appears to only happen for some points that appear in the batch _before_ the mismatched type point.

__System info:__ [Include InfluxDB version, operating system name, and other relevant details]

current master (commit 8c2cfd1)

__Steps to reproduce:__

1. Create database: `> create database telegraf`
2. Write `test` measurement:
```
$ curl -i "http://localhost:8086/write?db=telegraf" --data-binary "test value=1"
HTTP/1.1 204 No Content
Content-Type: application/json
Request-Id: e88b50c9-d74f-11e6-8009-000000000000
X-Influxdb-Version: v1.1.0
Date: Tue, 10 Jan 2017 16:14:48 GMT
```
3. Show measurements:
```
> show measurements
name: measurements
name
----
test
```
4. Write `test2` and `test3` measurements in a batch with a mismatched type `test` point at the end:
```
$ curl -i "http://localhost:8086/write?db=telegraf" --data-binary "test2 value=1                                                                
test3 value=1
test value=1i
"
HTTP/1.1 400 Bad Request
Content-Type: application/json
Request-Id: fc6afa4c-d74f-11e6-800b-000000000000
X-Influxdb-Version: v1.1.0
Date: Tue, 10 Jan 2017 16:15:21 GMT
Content-Length: 125

{"error":"field type conflict: input field \"value\" on measurement \"test\" is type integer, already exists as type float"}
```
5. Observe that `show measurements` now shows `test3`:
```
> show measurements
name: measurements
name
----
test
test3
```
6. But there is no data in `test3`, and it disappears on restart:
```
> select * from test3
>
```

__Additional info:__ [Include gist of relevant config, logs, etc.]

```
> show shards
name: _internal
id database  retention_policy shard_group start_time           end_time             expiry_time          owners
-- --------  ---------------- ----------- ----------           --------             -----------          ------
1  _internal monitor          1           2017-01-10T00:00:00Z 2017-01-11T00:00:00Z 2017-01-18T00:00:00Z


name: telegraf
id database retention_policy shard_group start_time           end_time             expiry_time          owners
-- -------- ---------------- ----------- ----------           --------             -----------          ------
3  telegraf autogen          3           2017-01-09T00:00:00Z 2017-01-16T00:00:00Z 2017-01-16T00:00:00Z

> show diagnostics
name: build
Branch Build Time Commit                                   Version
------ ---------- ------                                   -------
master            800da5732b91c816b0a097acf8887fa2af1efa1a v1.1.0


name: network
hostname
--------
tars


name: runtime
GOARCH GOMAXPROCS GOOS   version
------ ---------- ----   -------
amd64  8          darwin go1.7.3


name: system
PID  currentTime                    started                        uptime
---  -----------                    -------                        ------
6078 2017-01-10T16:16:52.058524797Z 2017-01-10T16:13:37.986046031Z 3m14.072478867s

```

**Comment by phemmer** [comment]
Hrm, I believe this behavior changed since I wrote https://github.com/influxdata/influxdb/issues/4856#issuecomment-238402453. I'm fairly certain that when I wrote that, the points in the batch before the error were actually inserted. Not sure what version that was though.

**Comment by sparrc** [comment]
closed by #7836
